rs/rust_canisters/random_traffic_test/src/main.rs

```diff
@@ -0,0 +1,236 @@
+use candid::{Decode, Encode};
+use dfn_core::api;
+use ic_base_types::CanisterId;
+use rand::{
+    distributions::{Distribution, WeightedIndex},
+    rngs::StdRng,
+    seq::SliceRandom,
+    Rng, SeedableRng,
+};
+use random_traffic_test::*;
+use std::cell::{Cell, RefCell};
+use std::collections::BTreeMap;
+use std::ops::RangeInclusive;
+
+thread_local! {
+    /// Random number generator used for determining payload sizes et.al.
+    static RNG: RefCell<StdRng> = RefCell::new(StdRng::seed_from_u64(13));
+    /// Weight for making a reply used in a weighted binomial distribution.
+    static REPLY_WEIGHT: Cell<u32> = const { Cell::new(1) };
+    /// Weight for making a downstream call used in a weighted binomial distribution.
+    static CALL_WEIGHT: Cell<u32> = const { Cell::new(0) };
+    /// A configuration holding parameters for how to the canister should behave, such as the range
+    /// of payload bytes it should send.
+    static CONFIG: RefCell<Config> = RefCell::default();
+    /// The maximum number of calls each heartbeat can make.
+    static MAX_CALLS_PER_HEARTBEAT: Cell<u32> = Cell::default();
+    /// An ID used as an entry in `RECORDS` unique for each call.
+    static CALL_ID: Cell<u32> = Cell::default();
+    /// A collection of records; one record for each call. Keeps track of how each call went,
+    /// whether it was rejected or not and how many bytes we received as a reply.
+    static RECORDS: RefCell<BTreeMap<u32, Record>> = RefCell::default();
+}
+
+/// Replaces the canister state according to `f` by parsing `arg_data` of type `T`; replies the old
+/// state.
+fn replace_state<T, F>(f: F)
+where
+    T: candid::CandidType + for<'a> serde::Deserialize<'a>,
+    F: FnOnce(T) -> T,
```

Comment:
What I mean by canister state is an element in the `thread_local {}` block, i.e. an element of the canister's state.

Reply:
"Canister state" has a widely used meaning outside of this context and one must make a conscious effort to ignore that when reading this doc comment. "Global" or "thread local" would be more accurate.

Although, as said, it is trivial to pass an `f` that doesn't replace anything (e.g. `replace(|_| print!("nothing"))`). So maybe something like `with_candid_encoding()` would be a more accurate name overall.

Reply:
> Although, as said, it is trivial to pass an f that doesn't replace anything (e.g. replace(|_: ()| print!("nothing"))). So maybe something like with_candid_encoding() would be a more accurate name overall.

I don't see a problem here since it does nothing, but also not harm either. I called it `with_candid_encoding()`.

Reply:
Please also update the doc comment.

Reply:
Done.